### ISSUE TITLE
Bug in the RequestBuffer constructor

### ISSUE THREAD

**Issue by sidit77** [kevinmehall/nusb#49]
Hey I've noticed that one of the constructors of `RequestBuffer` seems to contain an error:

```rust
pub fn reuse(v: Vec<u8>, len: usize) -> RequestBuffer {
    let mut v = ManuallyDrop::new(v);
    v.reserve_exact(len.saturating_sub(len));
    RequestBuffer {
        buf: v.as_mut_ptr(),
        capacity: v.capacity(),
        requested: len,
    }
}
```
`len.saturating_sub(len)` is always zero. I suspect this should be `len.saturating_sub(v.len())` instead.

**Comment by kevinmehall** [comment]
Good catch, thanks!

**Comment by kevinmehall** [comment]
Fix released in [v0.1.8](https://github.com/kevinmehall/nusb/releases/tag/v0.1.8)
